src/core/reactor.cc

```diff
@@ -4414,6 +4414,12 @@ void smp::configure(const smp_options& smp_opts, const reactor_options& reactor_
     std::optional<memory::internal::numa_layout> layout;
     if (smp_opts.memory_allocator == memory_allocator::seastar) {
         layout = memory::configure(allocations[0].mem, mbind, use_transparent_hugepages, hugepages_path);
+    } else {
+        // #2148 - if running seastar allocator but options that contradict this, we still need to
+        // init memory at least minimally, otherwise a bunch of stuff breaks.
+        // Previously, we got away wth this by accident due to #2137.
+        // Calling memory::configure without any memory etc will fix up internal structures enough.
+        memory::configure({}, false, false, std::nullopt);
```

Comment:
I just checked and this will set `is_reactor_thread=true`, which means that the seastar allocator will start serving allocation requests. This effectively breaks `smp_options::memory_allocator = memory_allocator::standard`, which scylla tools use.

Reply:
An alternative option is adding "config_minimal" that only does init_cpu_mem().

Reply:
Changed to the above.